client/src/client.rs

```diff
@@ -278,18 +278,54 @@ pub trait RpcApi: Sized {
         blank: Option<bool>,
         passphrase: Option<&str>,
         avoid_reuse: Option<bool>,
+        descriptors: Option<bool>,
     ) -> Result<json::LoadWalletResult> {
-        let mut args = [
-            wallet.into(),
-            opt_into_json(disable_private_keys)?,
-            opt_into_json(blank)?,
-            opt_into_json(passphrase)?,
-            opt_into_json(avoid_reuse)?,
-        ];
-        self.call(
-            "createwallet",
-            handle_defaults(&mut args, &[false.into(), false.into(), into_json("")?, false.into()]),
-        )
+        // the descriptors argument was added in version 21
+        if self.version()? < 210000 {
+            // note: we allow Some(false) since it's the default behavior
+            if let Some(true) = descriptors {
+                return Err(Error::Unsupported);
+            }
+            // no descriptors argument yet
+            let mut args = [
+                wallet.into(),
+                opt_into_json(disable_private_keys)?,
+                opt_into_json(blank)?,
+                opt_into_json(passphrase)?,
+                opt_into_json(avoid_reuse)?,
+            ];
+            self.call(
+                "createwallet",
+                handle_defaults(
+                    &mut args,
+                    &[false.into(), false.into(), into_json("")?, false.into()],
+                ),
+            )
+        } else {
+            let mut args = [
+                wallet.into(),
+                opt_into_json(disable_private_keys)?,
+                opt_into_json(blank)?,
+                opt_into_json(passphrase)?,
+                opt_into_json(avoid_reuse)?,
+                opt_into_json(descriptors)?,
+            ];
+            // from 23 on, the default value of the descriptors argument is true
+            let default_descriptors = self.version()? >= 230000;
```

Comment:
I'm a bit confused about the goal of this line. You are setting `default_descriptors` to whatever its default value would be ... so why provide it at all?

Reply:
It was easier to determine the default value than to use a variable number of arguments. Also there are more `createwallet` arguments in the newest version that are not added yet (`load_on_startup` and `external_signer`), and if they're added in the future, _some_ value will need to be passed here anyway, right?

Reply:
FWIW in theory, no, it's possible to use named arguments rather than positional ones, so that we don't have to stick values in just to get to further values. But that's a bigger change and unrelated to this PR :).

>It was easier to determine the default value than to use a variable number of arguments.

Okay, but simply dropping `default_descriptors` won't cause you to need a variable number of arguments. Right now you always pass it. I'm suggesting you never pass it.

Reply:
@apoelstra if I don't include `default_descriptors` in the list of default arguments, the defaults would be applied incorrectly, right? At least if I understand the documentation of `handle_defaults` correctly:
```rust
/// Note, that `defaults` corresponds to the last elements of `args`.
///
/// ```norust
/// arg1 arg2 arg3 arg4
///           def1 def2
```

Reply:
Handle defaults never touches the last argument anyway, so this will only become relevant if we add another argument here.

Also, on named arguments, my [total refactor for async support](https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/294) migrates to using named arguments so we don't need the ugly `handle_defaults` stuff anymore.

Reply:
To answer Andrew's concern:

> I'm a bit confused about the goal of this line. You are setting default_descriptors to whatever its default value would be ... so why provide it at all?

He's only setting the default value that is filled if the actual value is not set.